dtfs-central-api/src/v1/controllers/utilisation-report-service/get-utilisation-report-reconciliation-details-by-id.controller/helpers/get-utilisation-details.ts

```diff
@@ -0,0 +1,67 @@
+import { FeeRecordEntity, TfmFacility } from '@ukef/dtfs2-common';
+import Big from 'big.js';
+import { FeeRecordUtilisation } from '../../../../../types/fee-records';
+import { TfmFacilitiesRepo } from '../../../../../repositories/tfm-facilities-repo';
+import { NotFoundError } from '../../../../../errors';
+import { getLatestCompletedAmendmentToFacilityValue } from '../../../../../helpers';
+
+/**
+ * Calculates the exposure based on the current utilisation
+ * @param utilisation - The current utilisation of the facility
+ * @param coverPercentage - The cover percentage of the facility
+ * @returns The exposure
+ */
+export const calculateExposure = (utilisation: number, coverPercentage: number) => {
+  return new Big(utilisation).mul(coverPercentage).div(100).round(2).toNumber();
+};
+
+/**
+ * Retrieves and constructs utilisation data for the given fee records
+ * @param feeRecords - The fee records
+ * @returns Utilisation data for each fee record
+ */
+export const getUtilisationDetails = async (feeRecords: FeeRecordEntity[]): Promise<FeeRecordUtilisation[]> => {
+  const allFacilityIds = feeRecords.map((feeRecord) => feeRecord.facilityId);
+  const tfmFacilities = await TfmFacilitiesRepo.findByUkefFacilityIds(allFacilityIds);
+  const tfmFacilitiesRecord: Record<string, TfmFacility> = {};
+  tfmFacilities.forEach((tfmFacility) => {
+    const { ukefFacilityId } = tfmFacility.facilitySnapshot;
+    if (ukefFacilityId) {
+      tfmFacilitiesRecord[ukefFacilityId] = tfmFacility;
+    }
+  });
+
+  return feeRecords.map(
```

Comment:
Can we move this mapping into its own file/helper please i.e

```ts
const mapFeeRecord = (feeRecord) => {
  const { value, coverPercentage } = tfmFacility.facilitySnapshot;

  return {
    facilityd,
    exporter,
    baseCurrency,
    value: amendedValue ?? value,
    // etc
  };
};

const mapFeeRecords = (feeRecords) => {
  const mapped = feeRecords.map((record) => mapFeeRecord(record));

  return mapped;
};
```

Then the controller can become simply e.g `return mapFeeRecords();`